### PR TITLE
fix(cirriculum) - allow spaces in fill in blank challenges

### LINKED ISSUE #52842: Allow spaces in fill in blank challenges for the English section
### Describe the Issue

A lot of the fill in the blank challenges are set up in a way such that spacing really shouldn't matter. However due to the way it is set up, some campers are getting tripped up by spaces. 

### Affected Page

https://www.freecodecamp.org/learn/a2-english-for-developers/

### Your code

```



```

### Expected behavior

Spacing really shouldn't matter and as long as it is correctly cased , the camper should be able to proceed. 

### Screenshots

_No response_

### System

- Device: Desktop
- OS: Windows 10
- Browser: FireFox
- Version: 121.0

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #52842

This was what I was discussing in this thread on the FCC [forums](https://forum.freecodecamp.org/t/should-the-english-cirriculum-be-space-friendly/659802).

### CODE REVIEW

File: client/src/templates/Challenges/fill-in-the-blank/show.tsx [hunk 1]
```diff
@@ -180,8 +180,8 @@ class ShowFillInTheBlank extends Component<
 
     const blankAnswers = blanks.map(b => b.answer);
 
-    const newAnswersCorrect = userAnswers.map(
-      (userAnswer, i) => userAnswer === blankAnswers[i]
+    const newAnswersCorrect = userAnswers.map((userAnswer, i) =>
+      userAnswer != null ? userAnswer.trim() === blankAnswers[i] : false
```
Comment (naomi-lgbt) [inline client/src/templates/Challenges/fill-in-the-blank/show.tsx:184]:
```suggestion
    const newAnswersCorrect = userAnswers.map((userAnswer, i) =>
      userAnswer && userAnswer.trim() === blankAnswers[i]
```

We could skip the ternary and just assert truthiness.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: client/src/templates/Challenges/fill-in-the-blank/show.tsx [hunk 2]
```diff
@@ -181,7 +181,7 @@ class ShowFillInTheBlank extends Component<
     const blankAnswers = blanks.map(b => b.answer);
 
     const newAnswersCorrect = userAnswers.map((userAnswer, i) =>
-      userAnswer != null ? userAnswer.trim() === blankAnswers[i] : false
+      userAnswer && userAnswer.trim() === blankAnswers[i]
     );
```
Comment (naomi-lgbt) [inline client/src/templates/Challenges/fill-in-the-blank/show.tsx:185]: Can't provide a suggestion across a deleted line, but this needs to be:

```js
    const newAnswersCorrect = userAnswers.map(
      (userAnswer, i) => userAnswer && userAnswer.trim() === blankAnswers[i]
    );
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: client/src/templates/Challenges/fill-in-the-blank/show.tsx (hunk 2 above)
Comment (a2937) [reply client/src/templates/Challenges/fill-in-the-blank/show.tsx:185]: Wait Naomi. I just looked at my Visual Studio Code. There is an error message that wasn't there before I started my pull request. .

"Type '(boolean | "" | null)[]' is not assignable to type '(boolean | null)[]'." on line 200. The same error message is on line 190. I think that something unusual might end up being stored in the `newAnswersCorrect` variable. 

For some reason, Visual Studio Code believes that an empty string or null may be assigned to it. And given what I know of Javascript, that does seem pretty believable. While it is valid and falsy, it doesn't feel quite right in a situation where typings matter.

File: client/src/templates/Challenges/fill-in-the-blank/show.tsx [hunk 3]
```diff
@@ -180,8 +180,8 @@ class ShowFillInTheBlank extends Component<
 
     const blankAnswers = blanks.map(b => b.answer);
 
-    const newAnswersCorrect = userAnswers.map(
-      (userAnswer, i) => userAnswer === blankAnswers[i]
+    const newAnswersCorrect = userAnswers.map((userAnswer, i) =>
+      userAnswer && userAnswer.trim() === blankAnswers[i]
```
Comment (naomi-lgbt) [inline client/src/templates/Challenges/fill-in-the-blank/show.tsx:184]:
```suggestion
      !!userAnswer && userAnswer.trim() === blankAnswers[i]
```

Easy fix - we just boolean it.

Comment (a2937) [reply client/src/templates/Challenges/fill-in-the-blank/show.tsx]: Okay done.

**Review by naomi-lgbt — APPROVED** [review]

**Review by huyenltnguyen — APPROVED** [review]

**Pull request merged**